### PR TITLE
[MIG] purchase_last_price_info: Migration to 10.0

### PR DESCRIPTION
**Pull request by kittiu**
Propose to merge this module.

### CODE REVIEW

**Comment by kittiu** [comment]
Can this one be merged?
(I see the runbot failed but with no reason).

File: purchase_last_price_info/models/product.py [hunk 1]
```diff
@@ -14,16 +11,22 @@ def _get_last_purchase(self):
         """ Get last purchase price, last purchase date and last supplier """
         lines = self.env['purchase.order.line'].search(
             [('product_id', '=', self.id),
-             ('state', 'in', ['confirmed', 'done'])]).sorted(
+             ('state', 'in', ['purchase', 'done'])]).sorted(
             key=lambda l: l.order_id.date_order, reverse=True)
         self.last_purchase_date = lines[:1].order_id.date_order
         self.last_purchase_price = lines[:1].price_unit
         self.last_supplier_id = lines[:1].order_id.partner_id
```
Comment (Abranes) [inline purchase_last_price_info/models/product.py:18]: The `search` and `sorted` could be replaced by:
`self.env['purchase.order.line'].search(
            [('product_id', '=', self.id),
             ('state', 'in', ['purchase', 'done'])],
            order='date_order DESC', limit=1)`

Then you can change `lines[:1]` to `lines`.

Also you can take the `date_order` and `partner_id` from lines directly.

File: purchase_last_price_info/models/product.py [hunk 2]
```diff
@@ -14,16 +11,22 @@ def _get_last_purchase(self):
...
 
     last_purchase_price = fields.Float(
-        string='Last Purchase Price', compute='_get_last_purchase')
+        string='Last Purchase Price',
+        compute='_get_last_purchase',
+    )
```
Comment (Abranes) [inline purchase_last_price_info/models/product.py:23]: Add the `digits=dp.get_precision('Product Price')` for display the same decimal precisions as in the lines.

**Review by Abranes — CHANGES_REQUESTED** [review]

**Comment by Abranes** [comment]
Change `@api.one` for `@api.multi`.

For the rest, the module is working.

@kittiu if you can't do the changes I can do it.

**Comment by JayVora-SerpentCS** [comment]
ping @kittiu

File: purchase_last_price_info/__manifest__.py [hunk 3]
```diff
@@ -1,41 +1,22 @@
...
-    "version": "8.0.1.0.0",
+    "version": "10.0.1.0.0",
     "category": "Purchase Management",
     "license": "AGPL-3",
     "author": "OdooMRP team, "
               "AvanzOSC, "
               "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
+              "Ecosoft - Kitti U., "
               "Odoo Community Association (OCA)",
     "website": "http://www.odoomrp.com",
```
Comment (nikul-serpentcs) [inline purchase_last_price_info/__manifest__.py:13]: Replace website URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

File: purchase_last_price_info/models/product.py [hunk 4]
```diff
@@ -1,8 +1,5 @@
-# -*- encoding: utf-8 -*-
-##############################################################################
-# For copyright and license notices, see __openerp__.py file in root directory
-##############################################################################
-
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 from openerp import api, fields, models
```
Comment (nikul-serpentcs) [inline purchase_last_price_info/models/product.py:3]: odoo

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Improve Code

**Comment by pedrobaeza** [comment]
Superseeded by #587

**Pull request closed without merging**